code/controllers/master.dm

```diff
@@ -641,19 +641,18 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 	return tab_data
 
 /datum/controller/master/StartLoadingMap()
-	//disallow more than one map to load at once, multithreading it will just cause race conditions
-	while(map_loading)
-		stoplag()
-	for(var/S in subsystems)
-		var/datum/controller/subsystem/SS = S
-		SS.StartLoadingMap()
-	map_loading = TRUE
+	map_loading ++
+	if (map_loading == 1)
+		for(var/S in subsystems)
+			var/datum/controller/subsystem/SS = S
```

Comment:
Can you not put a typed loop with as() here?
(I know not your code but probably nice micro opt)

Reply:
I guess you could, micro opts are worthless on super cold paths like this though

Reply:
This is why I asked if you can make this typed